llvm/lib/Target/WebAssembly/WebAssemblyFastISel.cpp

```diff
@@ -839,9 +839,11 @@ bool WebAssemblyFastISel::selectCall(const Instruction *I) {
 
     unsigned Reg;
 
-    if (Attrs.hasParamAttr(I, Attribute::SExt))
+    if (Attrs.hasParamAttr(I, Attribute::SExt) ||
+        (IsDirect && Func->hasParamAttribute(I, Attribute::SExt)))
```

Comment:
IsDirect is true if the function call isn't an indirect call.

Reply:
I have been too long away from LLVM to LGTM, but I would guess that you would have a similar situation with indirect function calls as well.  Perhaps you could test a function that takes a function pointer parameter with sext / zext attributes.  But, again, perhaps this comment is off-base.

Reply:
Pointers are (on their way to being) opaque; I forget whether function pointers ever had these parameters attached, but in any case they soon won't, so for indirect calls I think it should just be the instruction.